projects/jupyter-server-ydoc/jupyter_server_ydoc/handlers.py

```diff
@@ -70,52 +70,71 @@ def create_task(self, aw):
         task.add_done_callback(self._background_tasks.discard)
 
     async def prepare(self):
-        if not self._websocket_server.started.is_set():
-            self.create_task(self._websocket_server.start())
-            await self._websocket_server.started.wait()
-
-        # Get room
-        self._room_id: str = room_id_from_encoded_path(self.request.path)
-
-        async with self._room_lock(self._room_id):
-            if self._websocket_server.room_exists(self._room_id):
-                self.room: YRoom = await self._websocket_server.get_room(self._room_id)
-            else:
-                if self._room_id.count(":") >= 2:
-                    # DocumentRoom
-                    file_format, file_type, file_id = decode_file_path(self._room_id)
-                    if file_id in self._file_loaders:
-                        self._emit(
-                            LogLevel.WARNING,
-                            None,
-                            "There is another collaborative session accessing the same file.\nThe synchronization between rooms is not supported and you might lose some of your changes.",
-                        )
-
-                    file = self._file_loaders[file_id]
-                    updates_file_path = f".{file_type}:{file_id}.y"
-                    ystore = self._ystore_class(path=updates_file_path, log=self.log)
-                    self.room = DocumentRoom(
-                        self._room_id,
-                        file_format,
-                        file_type,
-                        file,
-                        self.event_logger,
-                        ystore,
-                        self.log,
-                        self._document_save_delay,
-                    )
-
+        res = super().prepare()
+        if isinstance(res, Awaitable):
```

Comment:
is there ever a scenario where this isn't an awaitable? I guess `None`? 

Can we use `ensure_async` from jupyter server?

```python
from jupyter_server.utils import ensure_async

await ensure_async(super().prepare())
```